packages/alfa-table/src/rowgroup.ts

```diff
@@ -0,0 +1,251 @@
+    private readonly _width: number;
+    private readonly _cells: Array<Cell.Builder>;
+    private readonly _rowgroup: RowGroup;
+
+    public static of(
+      y: number,
+      height: number,
+      element: Element,
+      width: number = 0,
+      cells: Array<Cell.Builder> = []
+    ): Builder {
+      return new Builder(y, height, element, width, cells);
+    }
+
+    constructor(
+      y: number,
+      height: number,
+      element: Element,
+      width: number,
+      cells: Array<Cell.Builder>
+    ) {
+      this._rowgroup = RowGroup.of(y, height, element);
+      this._width = width;
+      this._cells = cells;
+    }
+
+    public update(update: {
+      y?: number;
+      width?: number;
+      height?: number;
+      element?: Element;
+      cells?: Array<Cell.Builder>;
+    }): Builder {
+      return Builder.of(
+        update.y !== undefined ? update.y : this._rowgroup.anchor.y,
+        update.height !== undefined ? update.height : this._rowgroup.height,
+        update.element !== undefined ? update.element : this._rowgroup.element,
+        update.width !== undefined ? update.width : this._width,
+        update.cells !== undefined ? update.cells : this._cells
+      );
+    }
+
+    public get rowgroup(): RowGroup {
+      return this._rowgroup;
+    }
+
+    public get width(): number {
+      return this._width;
+    }
+
+    public get cells(): Array<Cell.Builder> {
+      return this._cells;
+    }
+
+    public get anchor(): { y: number } {
+      return this._rowgroup.anchor;
+    }
+
+    public get height(): number {
+      return this._rowgroup.height;
+    }
+
+    public get element(): Element {
+      return this._rowgroup.element;
+    }
+
+    // anchoring a row group needs to move down all cells accordingly
+    public anchorAt(y: number): Builder {
+      return this.update({
+        y,
+        cells: this._cells.map((cell) =>
+          cell.anchorAt(cell.anchor.x, y + cell.anchor.y)
+        ),
+      });
+    }
+
+    public equals(value: unknown): value is this {
+      if (!(value instanceof Builder)) return false;
+      return (
+        this._rowgroup.equals(value._rowgroup) &&
+        this._width === value._width &&
+        this._cells.length === value._cells.length &&
+        this._cells.every((cell, idx) => cell.equals(value._cells[idx]))
+      );
+    }
+
+    public toJSON(): Builder.JSON {
+      return {
+        rowgroup: this._rowgroup.toJSON(),
+        width: this._width,
+        cells: this._cells.map((cell) => cell.cell.toJSON()),
+      };
+    }
+  }
+
+  export namespace Builder {
+    /**
+     * @see https://html.spec.whatwg.org/multipage/tables.html#algorithm-for-processing-row-groups
+     */
+    export function from(group: Element): Result<Builder, string> {
+      if (
+        group.name !== "tfoot" &&
+        group.name !== "tbody" &&
+        group.name !== "thead"
+      )
+        return Err.of("This element is not a row group");
```

Comment:
```suggestion
      if (
        group.name !== "tfoot" &&
        group.name !== "tbody" &&
        group.name !== "thead"
      ) {
        return Err.of("This element is not a row group");
      }
```

Make sure not to rely on indentation like this; space sensitive code, especially in languages where it's optional, is a foot-gun as I'm sure Apple can attest to: https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/ 🙈

Reply:
Shouldn't this be caught be linter/prettier? Maybe I just forgot to run it… 🙄

Reply:
ESLint has a rule that can catch it 👍